### PR TITLE
prevent duplicate attributes being generated by AQL queries

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

File: arangod/Indexes/Index.cpp [hunk 1]
```diff
@@ -513,6 +513,12 @@ double Index::selectivityEstimate(StringRef const* extra) const {
     estimate = _clusterSelectivity;
   }
 
+  if (estimate < 0.0) {
+    estimate = 0.0;
+  } else if (estimate > 1.0) {
+    estimate = 1.0;
+  }
+
   TRI_ASSERT(estimate >= 0.0 &&
```
Comment (graetzer) [inline arangod/Indexes/Index.cpp]: The assertion seems unnecessary now

File: arangod/Indexes/Index.cpp [hunk 2]
```diff
@@ -513,6 +513,12 @@ double Index::selectivityEstimate(StringRef const* extra) const {
     estimate = _clusterSelectivity;
   }
 
+  if (estimate < 0.0) {
```
Comment (graetzer) [inline arangod/Indexes/Index.cpp]: Why not max(0, min(1, estimate))

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2633/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2641/

**Pull request merged**